src/set_theory/cofinality.lean

```diff
@@ -330,19 +330,84 @@ induction_on a $ λ α r _, induction_on b $ λ β s _ b0, begin
         by injection h with h; congr; injection h } }
 end
 
-@[simp] theorem cof_cof (o : ordinal) : cof (cof o).ord= cof o :=
-le_antisymm (le_trans (cof_le_card _) (by simp)) $
-induction_on o $ λ α r _, by exactI
-let ⟨S, hS, e₁⟩ := ord_cof_eq r,
-    ⟨T, hT, e₂⟩ := cof_eq (subrel r S) in begin
-  rw e₁ at e₂, rw ← e₂,
-  refine le_trans (cof_type_le {a | ∃ h, (subtype.mk a h : S) ∈ T} (λ a, _)) ⟨⟨_, _⟩⟩,
-  { rcases hS a with ⟨b, bS, br⟩,
-    rcases hT ⟨b, bS⟩ with ⟨⟨c, cS⟩, cT, cs⟩,
-    exact ⟨c, ⟨cS, cT⟩, is_order_connected.neg_trans cs br⟩ },
-  { exact λ ⟨a, h⟩, ⟨⟨a, h.fst⟩, h.snd⟩ },
-  { exact λ ⟨a, ha⟩ ⟨b, hb⟩ h,
-      by injection h with h; congr; injection h },
+/-- A fundamental sequence for `a`, or FS for short, is an increasing sequence of length
+    `o = cof a` that converges at `a`. We provide `o` explicitly in order to avoid type rewrites. -/
+def is_fs (a o : ordinal.{u}) (f : Π b < o, ordinal.{u}) : Prop :=
```

Comment:
I would prefer a slightly more descriptive name.

Reply:
I saw the "FS" abbreviation on a certain wiki, but digging further I haven't been able to find it elsewhere. I'll spell this out in full instead.